setup/index.ts

```diff
@@ -1,11 +1,16 @@
-import { askSessionToken } from './prompt';
-import { get as getEnvValue, set as setEnvValue } from './env';
+import { /* askSessionToken, */ askIOBackendBaseurl } from "./prompt";
+import { get as getEnvValue, set as setEnvValue, EnvKey } from "../lib/env";
+
+const promptIfNot = async (key: EnvKey, fn: Function) => {
+  if (!getEnvValue(key)) {
+    const value = await fn();
```

Comment:
while it may be in this specific "restricted" case, generally speaking it's better to avoid callbacks when possible, they make things complex and hard to type :)

this kind of pattern(s) is better implemented with an [Option](https://github.com/gcanti/fp-ts/blob/1.12.0/src/Option.ts)
ie. `const myvalue = fromNullable(process.env[<myvar>]).getOrElseL(...)`

Reply:
I resolve this conversation as the code is outdated